src/include/duckdb/common/vector_operations/aggregate_executor.hpp

```diff
@@ -382,15 +382,74 @@ class AggregateExecutor {
 		}
 	}
 
+	using Frames = vector<FrameBounds>;
+
 	template <class STATE, class INPUT_TYPE, class RESULT_TYPE, class OP>
 	static void UnaryWindow(Vector &input, const ValidityMask &ifilter, AggregateInputData &aggr_input_data,
-	                        data_ptr_t state, const FrameBounds &frame, const FrameBounds &prev, Vector &result,
-	                        idx_t rid, idx_t bias) {
+	                        data_ptr_t state_p, const Frames &frames, Vector &result, idx_t rid) {
 
-		auto idata = FlatVector::GetData<const INPUT_TYPE>(input) - bias;
+		auto idata = FlatVector::GetData<const INPUT_TYPE>(input);
 		const auto &ivalid = FlatVector::Validity(input);
-		OP::template Window<STATE, INPUT_TYPE, RESULT_TYPE>(
-		    idata, ifilter, ivalid, aggr_input_data, *reinterpret_cast<STATE *>(state), frame, prev, result, rid, bias);
+		auto &state = *reinterpret_cast<STATE *>(state_p);
+		OP::template Window<STATE, INPUT_TYPE, RESULT_TYPE>(idata, ifilter, ivalid, aggr_input_data, state, frames,
+		                                                    result, rid);
+	}
+
+	template <typename OP>
+	static void IntersectFrames(const Frames &lefts, const Frames &rights, OP &op) {
+		const auto union_start = MinValue(rights[0].start, lefts[0].start);
+		const auto union_end = MaxValue(rights.back().end, lefts.back().end);
+		const FrameBounds last(union_end, union_end);
+
+		//	Subframe indices
+		idx_t l = 0;
+		idx_t r = 0;
+		for (auto i = union_start; i < union_end;) {
+			int overlap = 0;
```

Comment:
Nit: `idx_t`? There are more usages of `int` in this file.

Reply:
It's a two bit mask. I could make it a `uint8_t`?

Reply:
Sounds good to me!